### ISSUE TITLE
Passing templating engine context to child templates

### ISSUE THREAD

**Issue by pofider** [jsreport/jsreport-child-templates#1]
Something like this could work

``` js
function row() {
  return '{#child row @data.bookIndex=' + JSON.stringify(this) + '}';   
}
```

``` html
{{#each books}}
{{{row}}}
{{/each}}
```

Seems that the only trouble to make this working is the wrong regexp used for parsing child templates currently.

**Comment by bjrmatos** [comment]
we have another [user asking to support for this](https://forum.jsreport.net/topic/65/difficulties-passing-data-to-child-template-then-retrieving-it) 

so i think it is time to have it 😄

**Comment by pofider** [comment]
I've quickly took a look at this implementation, but the regexp gets messy when trying to match such stringified json. It needs a deeper look.

**Comment by itsjling** [comment]
Are there still any plans to support this? It'd be great to have.

In case anyone is curious, my current workaround is to pass a string to the child template which represents the key for the data. The template checks for the existence of this string and parses it to access the data at that key location.

**Comment by bjrmatos** [comment]
done as part of 04285843d39733583e8ce25913aa70469b0176f4

i ended up supporting this by using a custom syntax + helper because supporting with just regexp was impossible or in other words the final regexp would be too much complex.

so the way to use this is like this:

```js
function row() {
  return '{#child row @data.book$=' + childTemplateSerializeData(this) + '}';   
}
```

```
{{#each books}}
{{{row}}}
{{/each}}
```

NOTE: there are more examples that show usage in the tests

**Comment by bjrmatos** [comment]
this should be solved in `v1.1.0`
